src/System.Private.CoreLib/shared/System/Collections/IDictionaryEnumerator.cs

```diff
@@ -50,7 +51,7 @@ object Key
         // to GetValue with no intervening calls to GetNext will
         // return the same object.
         // 
-        object Value
+        object? Value
```

Comment:
It's a little strange to me that we've said DictionaryEntry.Key is nullable but IDictionaryEnumerator.Key is non-nullable.

Are there any built-in dictionaries in .NET that allow null keys?  If not, maybe we should annotate DictionaryEntry.Key as non-nullable (and the associated struct constructor argument), even though technically it can be null with `default(DictionaryEntry).Key`?

Reply:
Yes I agree. It is kind of strange. From what I searched there is not a Dictionary implementation that allow null keys in .NET, so I think it makes sense to follow that pattern, as here we want to transmit the intent.

Reply:
I've made `DictionaryEntry.Key` non-nullable as that is the intent usage of that class.